actors/evm/src/interpreter/instructions/memory.rs

```diff
@@ -53,6 +53,30 @@ pub fn get_memory_region(
     }))
 }
 
+#[inline]
+pub fn mcopy(
+    state: &mut ExecutionState,
+    _: &System<impl Runtime>,
+    dest_index: U256,
+    src_index: U256,
+    size: U256,
+) -> Result<(), ActorError> {
+    // We are copying between two potentially overlapping slices in the same memory.
+    // MCOPY spec: Copying takes place as if an intermediate buffer was used,
+    //             allowing the destination and source to overlap.
+
+    // expand memory to accomodate requested src_index + size
+    let region = get_memory_region(&mut state.memory, src_index, size)?.expect("empty region");
+    let memory_slice = state.memory[region.offset..region.offset + region.size.get()].to_vec();
+
+    // expand memory to match dest_index + size
+    let _destination_region =
+        get_memory_region(&mut state.memory, dest_index, size)?.expect("empty region");
+
+    //copy
+    copy_to_memory(&mut state.memory, dest_index, size, U256::zero(), &memory_slice, true)
```

Comment:
It would be nice if we did this at a slightly lower level to avoid the intermediate vec. Rust has a function for this: https://doc.rust-lang.org/std/primitive.slice.html#method.copy_within

Reply:
Basically, we'll have to duplicate/change the `copy_to_memory` function to `copy_within_memory`.

Reply:
Ok great, that makes sense!